spec/scenarios/customers/create_partner_and_run_billing_spec.rb

```diff
@@ -0,0 +1,191 @@
+  end
+
+  around { |test| lago_premium!(&test) }
+
+  it 'allows to switch customer to partner before customer has assigned plans' do
+    expect do
+      create_or_update_customer(
+        {
+          external_id: partner.external_id,
+          account_type: 'partner'
+        }
+      )
+      partner.reload
+    end.to change(partner, :account_type).from('customer').to('partner')
+      .and change(partner, :exclude_from_dunning_campaign).from(false).to(true)
+
+    create_subscription(
+      {
+        external_customer_id: partner.external_id,
+        external_id: partner.external_id,
+        plan_code: plan.code
+      }
+    )
+
+    expect do
+      create_or_update_customer(
+        {
+          external_id: partner.external_id,
+          account_type: 'customer'
+        }
+      )
+    end.not_to change(partner.reload, :account_type)
+  end
+
+  it 'creates partner-specific invoices without payments, with partner numbering, excluded from analytics' do
+    create_or_update_customer(
+      {
+        external_id: partner.external_id,
+        account_type: 'partner'
+      }
+    )
+
+    ### 24 Apr: Create subscriptions + charges.
+    apr24 = Time.zone.parse('2024-04-24')
+    travel_to(apr24) do
+      create(
+        :package_charge,
+        plan: plan,
+        billable_metric: metric,
+        pay_in_advance: false,
+        prorated: false,
+        invoiceable: true,
+        properties: {
+          amount: '2',
+          free_units: 1000,
+          package_size: 1000
+        }
+      )
+
+      customers.each do |customer|
+        create_subscription(
+          {
+            external_customer_id: customer.external_id,
+            external_id: customer.external_id,
+            plan_code: plan.code
+          }
+        )
+      end
+      create_subscription(
+        {
+          external_customer_id: partner.external_id,
+          external_id: partner.external_id,
+          plan_code: plan.code
+        }
+      )
+    end
+
+    ### 25 Apr: Ingest events for Plan 1.
+    apr24 = Time.zone.parse('2024-04-24')
+    travel_to(apr24) do
+      plan.subscriptions.each do |subscription|
+        create_event(
+          params.merge(
+            external_subscription_id: subscription.external_id
+          )
+        )
+      end
+      perform_all_enqueued_jobs
+    end
+
+    # May 1st: Billing run; check invoice numbering
+    may1 = Time.zone.parse('2024-05-1')
+    travel_to(may1) do
+      organization.update(created_at: 1.month.ago)
+      perform_billing
+      expect(organization.invoices.count).to eq(3)
+      expect(partner.invoices.count).to eq(1)
+
+      partner_invoice = partner.invoices.first
+      expect(partner_invoice.self_billed).to eq(true)
+      expect(partner_invoice.number).to eq("#{organization.document_number_prefix}-001-001")
+
+      organization_invoices = customers.map(&:invoices).flatten
+      expect(organization_invoices.map(&:self_billed).uniq).to eq([false])
```

Comment:
```suggestion
      expect(organization_invoices.map(&:self_billed)).not_to include(true)
```

Is this more expressive for you?